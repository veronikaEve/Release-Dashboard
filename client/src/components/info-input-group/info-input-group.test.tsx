import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import InfoInputGroup from "../info-input-group/info-input-group";

describe("InfoInputGroup", () => {
  const props = {
    label: "test label",
    inputOptions: ["test_branch_1", "test_branch_2", "test_branch_3"],
    setReleaseInfo: jest.fn,
  };

  test("should match snapshot", () => {
    const { asFragment } = render(<InfoInputGroup {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should display 'Choose a branch' by default", () => {
    render(<InfoInputGroup {...props} />);
    const combobox = screen.getByRole("combobox");
    expect(combobox).toHaveDisplayValue("Choose a branch");
  });

  test("should display selected value", async () => {
    render(<InfoInputGroup {...props} />);

    const combobox = screen.getByRole("combobox");
    const option2 = screen.getByRole("option", {
      name: "test_branch_2",
    });

    await userEvent.selectOptions(combobox, option2);

    expect(combobox).toHaveDisplayValue("test_branch_2");
  });
});
