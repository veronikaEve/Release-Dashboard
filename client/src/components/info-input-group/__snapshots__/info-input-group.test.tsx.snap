// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoInputGroup should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="info-group row"
  >
    <div
      class="info-group__label col"
    >
      test label
    </div>
    <div
      class="info-group__info col"
    >
      <select
        as="select"
        class="form-select"
      >
        <option>
          Choose a branch
        </option>
        <option
          value="test_branch_3"
        >
          test_branch_3
        </option>
        <option
          value="test_branch_2"
        >
          test_branch_2
        </option>
        <option
          value="test_branch_1"
        >
          test_branch_1
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
